src/android/java/com/pspdfkit/cordova/CordovaPdfActivity.java

```diff
@@ -14,96 +16,120 @@
-        if (fragment != null) {
-            fragment.removeDocumentListener(listener);
+        @Override
+        public void onDocumentSaveFailed(
+            @NonNull PdfDocument document, @NonNull Throwable exception) {
+          try {
+            final JSONObject data = new JSONObject();
+            data.put("message", exception.getMessage());
+            EventDispatcher.getInstance().sendEvent("onDocumentSaveFailed", data);
+          } catch (JSONException e) {
+            e.printStackTrace();
+          }
         }
-        currentActivity = null;
-    }
+      };
 
-    @Override
-    protected void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
-        bindActivity(this);
-    }
+  public static CordovaPdfActivity getCurrentActivity() {
+    return currentActivity;
+  }
 
-    @Override
-    protected void onDestroy() {
-        releaseActivity();
-        super.onDestroy();
+  private void bindActivity(@NonNull final CordovaPdfActivity activity) {
+    checkArgumentNotNull(activity, "activity");
+    if (currentActivity != null) {
+      throw new IllegalStateException(
+          "EventDispatcher only supports a single CordovaPdfActivity at a time.");
     }
-
-    @Override
-    public void finish() {
-        super.finish();
-        // Notify JavaScript listeners that the activity was dismissed.
-        EventDispatcher.getInstance().notifyActivityDismissed();
+    currentActivity = activity;
+    final PdfFragment pdfFragment = currentActivity.getPdfFragment();
+    if (pdfFragment == null) {
+      throw new IllegalStateException(
+          "EventDispatcher only supports binding to activities that have a fragment instance.");
     }
+    pdfFragment.addDocumentListener(listener);
+  }
 
-    /** Dismisses this PDF activity. */
-    public void dismiss() {
-        finish();
+  private void releaseActivity() {
+    PdfFragment fragment = currentActivity.getPdfFragment();
+    if (fragment != null) {
+      fragment.removeDocumentListener(listener);
     }
-
-    public boolean saveDocument() throws IOException {
-        final PdfDocument document = getDocument();
-        if (document != null) {
-            return document.saveIfModified();
-        } else {
-            return false;
-        }
+    currentActivity = null;
+  }
+
+  /**
+   * Adds given {@link Disposable} to disposable collection which will be automatically disposed as
+   * a part of activity lifecycle.
+   *
+   * @param disposable to add to the disposable collection.
+   */
+  public void addSubscription(Disposable disposable) {
+    compositeDisposable.add(disposable);
+  }
+
+  private void disposeSubscriptions() {
+    compositeDisposable.dispose();
+  }
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    bindActivity(this);
+  }
+
+  @Override
+  protected void onStop() {
+    disposeSubscriptions();
```

Comment:
This will cause disposal of subscriptions upon a configuration change. I don't think that this is what we want, since it might cause unexpected behavior in edge case scenarios.

What kind of subscriptions are added to this composite disposable? Are these tied to the UI lifecycle and should be stopped whenever the UI is recreated (i.e. during configuration changes) or should they be only stopped when the activity is being finished?

If finishing is what we are looking for, we should go with something like this: https://stackoverflow.com/a/9621078/2134962